ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -151,52 +152,78 @@ public TransactionSimulator(
         callParams.getFrom() != null ? callParams.getFrom() : DEFAULT_FROM;
-    final Account sender = publicWorldState.get(senderAddress);
-    final long nonce = sender != null ? sender.getNonce() : 0L;
-    final long gasLimit =
-        callParams.getGasLimit() >= 0 ? callParams.getGasLimit() : header.getGasLimit();
-    final Wei gasPrice = callParams.getGasPrice() != null ? callParams.getGasPrice() : Wei.ZERO;
-    final Wei value = callParams.getValue() != null ? callParams.getValue() : Wei.ZERO;
-    final Bytes payload = callParams.getPayload() != null ? callParams.getPayload() : Bytes.EMPTY;
+
+    BlockHeader blockHeaderToProcess = header;
 
     if (transactionValidationParams.isAllowExceedingBalance()) {
       updater.getOrCreate(senderAddress).getMutable().setBalance(Wei.of(UInt256.MAX_VALUE));
+      if (header.getBaseFee().isPresent()) {
+        blockHeaderToProcess =
+            BlockHeaderBuilder.fromHeader(header)
+                .baseFee(0L)
+                .blockHeaderFunctions(protocolSpec.getBlockHeaderFunctions())
+                .buildBlockHeader();
+      }
     }
 
-    final ProtocolSpec protocolSpec = protocolSchedule.getByBlockNumber(header.getNumber());
+    final Account sender = publicWorldState.get(senderAddress);
+    final long nonce = sender != null ? sender.getNonce() : 0L;
+    final Wei gasPrice = callParams.getGasPrice() != null ? callParams.getGasPrice() : Wei.ZERO;
+    final long gasLimit =
+        callParams.getGasLimit() >= 0
+            ? callParams.getGasLimit()
+            : blockHeaderToProcess.getGasLimit();
+    final Wei value = callParams.getValue() != null ? callParams.getValue() : Wei.ZERO;
+    final Bytes payload = callParams.getPayload() != null ? callParams.getPayload() : Bytes.EMPTY;
 
     final MainnetTransactionProcessor transactionProcessor =
-        protocolSchedule.getByBlockNumber(header.getNumber()).getTransactionProcessor();
+        protocolSchedule
+            .getByBlockNumber(blockHeaderToProcess.getNumber())
+            .getTransactionProcessor();
 
     final Transaction.Builder transactionBuilder =
         Transaction.builder()
             .nonce(nonce)
-            .gasPrice(gasPrice)
             .gasLimit(gasLimit)
             .to(callParams.getTo())
             .sender(senderAddress)
             .value(value)
             .payload(payload)
             .signature(FAKE_SIGNATURE);
-    callParams.getMaxPriorityFeePerGas().ifPresent(transactionBuilder::maxPriorityFeePerGas);
-    callParams.getMaxFeePerGas().ifPresent(transactionBuilder::maxFeePerGas);
+
+    if (header.getBaseFee().isEmpty()) {
+      transactionBuilder.gasPrice(gasPrice);
+    } else if (protocolSchedule.getChainId().isPresent()) {
+      transactionBuilder
+          .maxFeePerGas(callParams.getMaxFeePerGas().orElse(gasPrice))
+          .maxPriorityFeePerGas(callParams.getMaxPriorityFeePerGas().orElse(gasPrice));
+    } else {
+      return Optional.empty();
+    }
```

Comment:
is there a protocol schedule where we would have a base fee but no chain id?

Reply:
I removed this part because I found that the chainId is already set here 
https://github.com/hyperledger/besu/pull/2445/files#diff-49639cb13fe67367ebb91b4b732316588dd971719829ba51071141026b7888a9L188

It was useless to do that on this part

To answer the question, no it is not possible regarding the spec